File: java_projects/glue_code/joda-money/src/main/java/org/joda/money/format/LiteralPrinterParser.java
package org.joda.money.format;

import java.io.IOException;
import java.io.Serializable;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;
import org.joda.money.BigMoney;
import org.joda.money.ContextInitializer;
import org.joda.money.ExceptionHandler;

final class LiteralPrinterParser implements MoneyPrinter, MoneyParser, Serializable {
  private static final long serialVersionUID = 1L;
  private static Value clz =
      ContextInitializer.getPythonClass("/format/LiteralPrinterParser.py", "LiteralPrinterParser");
  private Value obj;

  public LiteralPrinterParser(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public String toString() {
    //
    // return "'" + literal + "'";
    //

    // TODO: Check the type mapping below!
    return this.obj.invokeMember("toString").as(String.class);
  }

  public void parse(MoneyParseContext context) {
    //
    // int endPos = context.getIndex() + literal.length();
    // if (endPos <= context.getTextLength()
    // && context.getTextSubstring(context.getIndex(), endPos).equals(literal)) {
    // context.setIndex(endPos);
    // } else {
    // context.setError();
    // }
    //

    this.obj.invokeMember("parse", context);
  }

  public void print(MoneyPrintContext context, Appendable appendable, BigMoney money)
      throws IOException {
    try {
      //
      // appendable.append(literal);
      //

      this.obj.invokeMember("print", context, appendable, money);
    } catch (PolyglotException e) {
      // TODO: Handle IOException
      throw (IOException) ExceptionHandler.handle(e, "LiteralPrinterParser.print");
    }
  }

  LiteralPrinterParser(String literal) {
    //
    // this.literal = literal;
    //

    this.obj = clz.invokeMember("__init__", literal);
  }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/joda-money/src/main/org/joda/money/format/LiteralPrinterParser.py
from __future__ import annotations
from io import IOBase
import io
from src.main.org.joda.money.BigMoney import *
from src.main.org.joda.money.format.MoneyParseContext import *
from src.main.org.joda.money.format.MoneyParser import *
from src.main.org.joda.money.format.MoneyPrintContext import *
from src.main.org.joda.money.format.MoneyPrinter import *


class LiteralPrinterParser(MoneyParser, MoneyPrinter):

    __literal: str = None

    __serialVersionUID: int = 1

    def toString(self) -> str:

        return "'" + self.__literal + "'"

    def parse(self, context: MoneyParseContext) -> None:

        end_pos = context.getIndex() + len(self.__literal)
        if (
            end_pos <= context.getTextLength()
            and context.getTextSubstring(context.getIndex(), end_pos) == self.__literal
        ):
            context.setIndex(end_pos)
        else:
            context.setError()

    def print(
        self, context: MoneyPrintContext, appendable: io.TextIOBase, money: BigMoney
    ) -> None:

        appendable.write(self.__literal)

    def __init__(self, literal: str) -> None:

        self.__literal = literal
